pkg/module/go_get_fetcher.go

```diff
@@ -169,6 +169,7 @@ func PrepareEnv(gopath string) []string {
 	cacheEnv := fmt.Sprintf("GOCACHE=%s", filepath.Join(gopath, "cache"))
 	gitSSH := fmt.Sprintf("GIT_SSH=%s", os.Getenv("GIT_SSH"))
 	gitSSHCmd := fmt.Sprintf("GIT_SSH_COMMAND=%s", os.Getenv("GIT_SSH_COMMAND"))
+	sshAuthSock := fmt.Sprintf("SSH_AUTH_SOCK=%s", os.Getenv("SSH_AUTH_SOCK"))
```

Comment:
@dfinkel can you check for the existence of `SSH_AUTH_SOCK` before adding this to the `cmdEnv` slice?

Also, are there any situations you can think of where someone would explicitly not want to use `SSH_AUTH_SOCK`?

Reply:
I've switched it to verify existence of the environment variable (and also added a check to verify that the file it points at both exists and is a unix socket (taking into account symlinks))

I don't know of a case where someone would have a valid `SSH_AUTH_SOCK` and wouldn't want to use it. It strictly provides better control, and openssh is generally smart enough to fall back to public key files and password auth if that doesn't work.

One can always unset the variable from the environment if they don't want it to propagate.

Reply:
@dfinkel ok, that sounds good. I am going to merge this, but will you be able to add a small section to the [configuring authentication documentation](https://docs.gomods.io/configuration/authentication/) about this env var?

Reply:
Sure, I'll send a PR in the next couple days to extend the documentation.